pyproject.toml

```diff
@@ -44,9 +44,13 @@ doc = [
     "sphinx",
     "pydata-sphinx-theme",
     "sphinx-gallery",
+    "sphinx-design",
     "matplotlib",
     "pillow",  # to scrape images from the examples
     "numpydoc",
+    "pycox",
+    "torch",
```

Comment:
We only need pycox to fetch some csv for the examples in the documentation. Should we bring these datasets on OpenML instead?
Also, the funny thing is:
- pycox requires torchtuples
- torchtuples raises an error when torch is not installed, but doesn't require it
cc @GaelVaroquaux @ogrisel